### DISCUSSION TITLE
New variable {{file_uri}} / {{event_file_uri}}

### DISCUSSION THREAD

**Discussion by catfist** [Taitava/obsidian-shellcommands#256] [Ideas - done & released]
Now I want to get url to open current note, to share that to external application (like Ticktick).
I tried to do it with `{{file_name}}` & cmd, but it's hard work.

On Linux or Mac, it wouldn't be too hard...

**Comment by Taitava** [comment]
Do you mean url for Obsidian URI or Obsidian Publish?

**Reply by catfist** [reply]
I mean build-in Obsidian URL, like `obsidian://open?vault=my%20vault&file=my%20note.md`.

https://help.obsidian.md/Advanced+topics/Using+obsidian+URI

**Reply by Taitava** [reply]
Yea. I think this would make sense. 🙂 

An alternative approach could be to offer URL encoding to current variables. Then you could use something like `obsidian://open?vault=my%20vault&file={{urLencoded_file_name}}`, but this contains more manual typing than what you suggested. The `urLencoded_` prefix is not decided yet, it will probably change to something else. Here's my older thoughts about url encoding variables: https://github.com/Taitava/obsidian-shellcommands/discussions/95#discussion-3669516 (see the _For the future: URL encoding_ section).

Of course different solutions can exist together. Your idea would provide a very easy to use way to create complete Obsidian URIs.

Thank you for your suggestion! 🌞

**Reply by catfist** [reply]
That's excellent! URL encoding serves my purpose well.
In that case, I think the new `{{vault_name}}` variable makes easier to construct URL.

**Reply by Taitava** [reply]
The URL encoding thing has  been on my mind for months, and I'd say your suggestion for new URL variables can be done sooner than the URL encoding thing. The reason is that for URL encoding, I need to decide a syntax for it (how the user should indicate that they want a variable value to be url encoded?) and it will also need some development on the variable system itself.

I will implement your new variable ideas before going into the more heavier URL encoding thing.

**Reply by catfist** [reply]
Of course, it's better for me! i'm looking forward to the update.

**Comment by Taitava** [comment]
# Small edit to the discussion's title
Old title: _New variable {{file_url}} / {{event_file_url}}_
New title: _New variable {{file_uri}} / {{event_file_uri}}_

I decided to use the word `uri` instead of `url` in the variable names, because [Obsidian URI](https://help.obsidian.md/Advanced+topics/Using+obsidian+URI) uses that word.

Also, I just wrote the code for these variables. They will be available in version `0.15.0`. 👍

**Reply by Taitava** [reply]
I created issue #258 that tracks the commits / releasing of this feature.

**Comment by Taitava** [comment]
# It's now released

Please let me know if it works for you or if there's anything to improve. 🙂

**Reply by catfist** [reply]
It worked perfectly! Thanks for the very quick update.
Example: I made command to add current note to TickTick with webhook & IFTTT

```sh
curl -X POST -H "Content-Type: application/json" -d "{\"value1\":\"[{{!title}}]({{!file_uri}})\",\"value2\":\"$today\",\"value3\":\"\"}" https://maker.ifttt.com/trigger/task_posted/with/key/xxxxxxxxxxxxxxxxxxx
```

**Reply by Taitava** [reply]
Thank you for your feedback! 🙂 Nice example! 🌞
